src/seedu/addressbook/AddressBook.java

```diff
@@ -36,1138 +36,1140 @@
-     */
-    private static ArrayList<String> splitByWhitespace(String toSplit) {
-        return new ArrayList<>(Arrays.asList(toSplit.trim().split("\\s+")));
-    }
+			showToUser(String.format(MESSAGE_INVALID_FILE, filePath));
+			exitProgram();
+		}
+	}
+
+	/**
+	 * Displays the goodbye message and exits the runtime.
+	 */
+	private static void exitProgram() {
+		showToUser(MESSAGE_GOODBYE, DIVIDER, DIVIDER);
+		System.exit(0);
+	}
+
+	/**
+	 * Sets up the storage based on the default file.
+	 * Creates file if missing.
+	 * Exits program if the file cannot be created.
+	 */
+	private static void setupDefaultFileForStorage() {
+		showToUser(MESSAGE_USING_DEFAULT_FILE);
+		storageFilePath = DEFAULT_STORAGE_FILEPATH;
+		createFileIfMissing(storageFilePath);
+	}
+
+	/**
+	 * Returns true if the given file path is valid.
+	 * A file path is valid if it has a valid parent directory as determined by {@link #hasValidParentDirectory}
+	 * and a valid file name as determined by {@link #hasValidFileName}.
+	 */
+	private static boolean isValidFilePath(String filePath) {
+		if (filePath == null) {
+			return false;
+		}
+		Path filePathToValidate;
+		try {
+			filePathToValidate = Paths.get(filePath);
+		} catch (InvalidPathException ipe) {
+			return false;
+		}
+		return hasValidParentDirectory(filePathToValidate) && hasValidFileName(filePathToValidate);
+	}
+
+	/**
+	 * Returns true if the file path has a parent directory that exists.
+	 */
+	private static boolean hasValidParentDirectory(Path filePath) {
+		Path parentDirectory = filePath.getParent();
+		return parentDirectory == null || Files.isDirectory(parentDirectory);
+	}
+
+	/**
+	 * Returns true if file path has a valid file name.
+	 * File name is valid if it has an extension and no reserved characters.
+	 * Reserved characters are OS-dependent.
+	 * If a file already exists, it must be a regular file.
+	 */
+	private static boolean hasValidFileName(Path filePath) {
+		return filePath.getFileName().toString().lastIndexOf('.') > 0
+				&& (!Files.exists(filePath) || Files.isRegularFile(filePath));
+	}
+
+	/**
+	 * Initialises the in-memory data using the storage file.
+	 * Assumption: The file exists.
+	 */
+	private static void loadDataFromStorage() {
+		initialiseAddressBookModel(loadPersonsFromFile(storageFilePath));
+	}
+
+
+	/*
+	 * ===========================================
+	 *           COMMAND LOGIC
+	 * ===========================================
+	 */
+
+	/**
+	 * Executes the command as specified by the {@code userInputString}
+	 *
+	 * @param userInputString  raw input from user
+	 * @return  feedback about how the command was executed
+	 */
+	private static String executeCommand(String userInputString) {
+		final String[] commandTypeAndParams = splitCommandWordAndArgs(userInputString);
+		final String commandType = commandTypeAndParams[0];
+		final String commandArgs = commandTypeAndParams[1];
+		switch (commandType) {
+		case COMMAND_ADD_WORD:
```

Comment:
the 'case' must be indented in the 'switch' class. Take a look at the coding standards once more.